test/utils/wasm_engine.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "types.hpp"
+#include <memory>
+#include <optional>
+
+namespace fizzy::test
+{
+/// The stateful representation of a wasm engine for testing purposes.
+class WasmEngine
+{
+public:
+    struct Result
+    {
+        bool trapped = false;
+        std::optional<uint64_t> value;
+    };
+
+    virtual ~WasmEngine() noexcept = default;
+
+    /// Parses input bytes and creates internal wasm module out of it.
+    /// Consecutive invokes replace the internal module and invalidate the internal instance.
```

Comment:
Did you actually mean "invalidate the internal instance" ? FizzyEngine doesn't seem to do that.

Reply:
Yes. There is nothing to do here. It means you are not allowed to use it any more and you need to call `instantiate()` again.

Reply:
Right. We agreed on in that in a previous comment.